Mathlib/Algebra/Order/Monoid/Units.lean

```diff
@@ -0,0 +1,57 @@
+/-
+Copyright (c) 2016 Jeremy Avigad. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jeremy Avigad, Leonardo de Moura, Mario Carneiro, Johannes Hölzl
+-/
+import Mathlib.Order.Hom.Basic
+import Mathlib.Order.MinMax
+import Mathlib.Algebra.Group.Units
+
+/-!
+# Units in ordered monoids
+-/
+
+
+namespace Units
+
+@[to_additive]
+instance [Monoid α] [Preorder α] : Preorder αˣ :=
+  Preorder.lift val
+
+@[simp, norm_cast, to_additive]
+theorem val_le_val [Monoid α] [Preorder α] {a b : αˣ} : (a : α) ≤ b ↔ a ≤ b :=
+  Iff.rfl
+#align units.coe_le_coe Units.val_le_val
+
+@[simp, norm_cast, to_additive]
+theorem val_lt_val [Monoid α] [Preorder α] {a b : αˣ} : (a : α) < b ↔ a < b :=
+  Iff.rfl
+#align units.coe_lt_coe Units.val_lt_val
+
+@[to_additive]
+instance [Monoid α] [PartialOrder α] : PartialOrder αˣ :=
+  PartialOrder.lift val Units.ext
+
+@[to_additive]
+instance [Monoid α] [LinearOrder α] : LinearOrder αˣ :=
+  LinearOrder.lift' val Units.ext
+
+/-- `val : αˣ → α` as an order embedding. -/
+@[to_additive "`val : add_units α → α` as an order embedding.",
+  simps (config := { fullyApplied := false })]
+def orderEmbedding_val [Monoid α] [LinearOrder α] : αˣ ↪o α :=
```

Comment:
Should this be `orderEmbeddingVal`? I don't think there should be underscores in a `def`